src/parakeet/UdpSocket.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+	Copyright 2021 OpenJAUS, LLC (dba MechaSpin). Subject to the MIT license.
+*/
+
+#include <parakeet/UdpSocket.h>
+
+#include <iostream>
+
+#if defined(_WIN32)
+#include <WS2tcpip.h>
+#include <winsock2.h>
+
+#pragma comment(lib, "ws2_32.lib")
+#elif defined(__linux) || defined(linux) || defined(__linux__)
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#endif
+
+namespace mechaspin
+{
+namespace parakeet
+{
+	const int MAX_BUFFER_LENGTH = 8192;
+	const int MAX_IP_LENGTH = 200;
+
+	#if defined(_WIN32)
+		WSADATA wsaData;
+	#endif
+
+	bool UdpSocket::open(int srcPort)
+	{
+		#if defined(_WIN32)
+			WSAStartup(MAKEWORD(2, 2), &wsaData);
+		#endif
+
+		socket = ::socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+
+		if (socket == -1)
+		{
+			return false;
+		}
+
+		sockaddr_in addr;
+		addr.sin_family = AF_INET;
+		addr.sin_port = htons(srcPort);
+		addr.sin_addr.s_addr = htonl(INADDR_ANY);
+
+		if (::bind(socket, (struct sockaddr*)&addr, sizeof(addr)) != 0)
+		{
+			return false;
+		}
+
+		return true;
+	}
+
+	void UdpSocket::close()
+	{
+		if (isConnected())
+		{
+			#if defined(_WIN32)
+				closesocket(socket);
+				WSACleanup();
+			#elif defined(__linux) || defined(linux) || defined(__linux__)
+				::close(socket);
+			#endif
+			socket = 0;
+		}
+	}
+
+	int UdpSocket::read(const mechaspin::parakeet::internal::BufferData& bufferData, int bufferMaxSize)
```

Comment:
We probably want to also pass the source InetAddress back to the user as well. Our driver code currently doesn't care but as a stand-alone class, it would be be necessary.

Reply:
YT issue created.